scene/gui/code_edit.cpp

```diff
@@ -1756,6 +1806,74 @@ TypedArray<int> CodeEdit::get_folded_lines() const {
 	return folded_lines;
 }
 
+/* Code region */
+void CodeEdit::create_code_region() {
+	// Abort if there is no selected text
+	if (!has_selection()) {
+		return;
+	}
+	// Check that region tag find a comment delimiter and is valid
+	if (code_region_start.is_empty()) {
+		WARN_PRINT_ONCE("Cannot create code region without any one line comment delimiters");
+		return;
+	}
+	begin_complex_operation();
+	// Adding start and end region tags
+	int first_region_start = -1;
+	for (int caret_idx : get_caret_index_edit_order()) {
```

Comment:
There's a edge case here if there are overlapping selections like this:

![image](https://github.com/godotengine/godot/assets/6584330/78828aee-2294-46f5-803a-fdd055d623dc)

In the other multi-caret code, we'd combine them into one region, we should probably do something similar here.

Reply:
```// Merge selections if next selection starts on the same line the previous one ends
	Vector<int> caret_edit_order = get_caret_index_edit_order();
	for (int i = 1; i < caret_edit_order.size(); i++) {
		int current_caret = caret_edit_order[i - 1];
		int next_caret = caret_edit_order[i];
		if (get_selection_from_line(current_caret) == get_selection_to_line(next_caret)) {
			select(get_selection_from_line(next_caret), get_selection_from_column(next_caret), get_selection_to_line(current_caret), get_selection_to_column(current_caret), next_caret);
			remove_caret(current_caret);
		}
	}
```

Trying to implents that. This code does the job, but not sure it's the good way to do that. Is there a better way ?

Reply:
Can do something like below, or take a look at `void CodeTextEditor::move_lines_up()`:
```
	Vector<int> caret_edit_order = get_caret_index_edit_order();
	Vector<int> carets_to_remove;
	for (int i = 0; i < caret_edit_order.size(); i++) {
		int current_caret = caret_edit_order[i];
		
		int to_line = get_selection_to_line(i, c);
		int from_line = get_selection_from_line(i, c);
	
		// Find next non-overlapping caret.		
		for (int j = i; j < caret_edit_order.size(); j++) {
			if (!has_selection(j)) {
				if (get_caret_column(j) != from_line) {
					break;
				}
				carets_to_remove.append(j);
				i++;
				continue;
			}
			
			if (get_selection_to_line(j) == from_line) {
				from_line = get_selection_to_line(j);
				carets_to_remove.append(j);
				i++;
				continue;
			}
			break;
		}
		
		// handle region
	}
	
	// Sort and remove backwards to preserve indexes.
	carets_to_remove.sort();
	for (int i = carets_to_remove.size() - 1; i >= 0; i--) {
		remove_caret(carets_to_remove[i]);
	}
```

Reply:
My implementation works in all case I can test. It seems to be sufficient here and more simple to understand.
Do you see any problem with it ?